### PR TITLE
Ref: Add instruction_addr  to SentryStackFrame

### PR DESCRIPTION
**Pull request by lucas-zimerman**
https://develop.sentry.dev/sdk/event-payloads/stacktrace/
instruction_addr  was missing from SentryStackFrame and it's an important parameter for Native exceptions (allowing it to get symbolicated)

### CODE REVIEW

File: src/Sentry/Protocol/Exceptions/SentryStackFrame.cs [hunk 1]
```diff
@@ -248,6 +257,13 @@ public void WriteTo(Utf8JsonWriter writer)
                 writer.WriteNumber("symbol_addr", symbolAddress);
             }
 
+            // Instruction address
+            if (InstructionAddress is { } instructionAddress &&
+                Regex.IsMatch(instructionAddress, @"\A\b0[xX][0-9a-fA-F]+\b\Z"))
```
Comment (bruno-garcia) [inline src/Sentry/Protocol/Exceptions/SentryStackFrame.cs:262]: if it doesn't match we just don't do anything? Should we match on the setter then?
Should this be a `long` instead?

File: src/Sentry/Protocol/Exceptions/SentryStackFrame.cs [hunk 2]
```diff
@@ -248,6 +257,13 @@ public void WriteTo(Utf8JsonWriter writer)
...
+            {
+                writer.WriteString("instruction_addr", instructionAddress.ToLower());
+            }
```
Comment (bruno-garcia) [inline src/Sentry/Protocol/Exceptions/SentryStackFrame.cs:265]: I'd rather we dont' have such logic buried in the serialization code.

We can add that elsewhere, or nowhere really (let it hit the server and the error there will alert the user it was malformed. That will only affect users creating frames "by hand" and doing it wrong, not us. (Assuming we get it right :) )

Also, why `ToLower`? it'll allocate a string. Can we just make sure we get the value in its right format into the object?

```suggestion
            if (!string.IsNullOrWhiteSpace(instructionAddress))
            {
                writer.WriteString("instruction_addr", instructionAddress);
            }
```

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

File: src/Sentry/Protocol/Exceptions/SentryStackFrame.cs (hunk 2 above)
Comment (lucas-zimerman) [reply src/Sentry/Protocol/Exceptions/SentryStackFrame.cs]: Alright, got rid of that code :D

File: src/Sentry/Protocol/Exceptions/SentryStackFrame.cs [hunk 3]
```diff
@@ -248,6 +257,12 @@ public void WriteTo(Utf8JsonWriter writer)
                 writer.WriteNumber("symbol_addr", symbolAddress);
             }
 
+            // Instruction address
+            if (InstructionAddress is { } instructionAddress)
```
Comment (bruno-garcia) [inline src/Sentry/Protocol/Exceptions/SentryStackFrame.cs:261]: This doesn't check if it's `""` or `"  "`, why did you chose this vs string.IsNullOrEmpty?

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**